linera-base/src/crypto/mod.rs

```diff
@@ -71,6 +72,15 @@ pub enum AccountSecretKey {
     Secp256k1(secp256k1::Secp256k1SecretKey),
 }
 
+impl Clone for AccountSecretKey {
+    fn clone(&self) -> Self {
+        match self {
+            AccountSecretKey::Ed25519(secret) => AccountSecretKey::Ed25519(secret.copy()),
+            AccountSecretKey::Secp256k1(secret) => AccountSecretKey::Secp256k1(secret.copy()),
+        }
+    }
+}
+
```

Comment:
It was a deliberate choice not to provide `Clone` for secret keys in the codebase.  I don't have strong opinions about it either way, but I'd suggest having a discussion.

Reply:
I know but if it wasn't needed by the _already-existing_ code I wouldn't have added it. Right now I'm just trying to get a green CI with the new abstractions but without changing anything else.

Reply:
I'm a bit confused how it could be needed by the existing code, since as far as I know there previously wasn't such an implementation for `AccountSecretKey` (and I can see you've removed some `copy()` calls below that are how we used to copy these).

Reply:
We used to `.copy()` [here](https://github.com/linera-io/linera-protocol/blob/main/linera-core/src/client/mod.rs#L1025).

Reply:
This `Clone` impl is now removed.